File: src/main/java/com/baulsupp/oksocial/services/twitter/twurlrc/TwurlRc.java
package com.baulsupp.oksocial.services.twitter.twurlrc;

import com.baulsupp.oksocial.services.twitter.TwitterCredentials;
import java.util.List;
import java.util.Map;

public class TwurlRc {
  public Map<String, Map<String, TwitterCredentials>> profiles;
  public Map<String, List<String>> configuration;

  public List<String> defaultProfile() {
    return configuration.get("default_profile");
  }

  public TwitterCredentials readCredentials(String username, String consumerKey) {
    return profiles.get(username).get(consumerKey);
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/twitter/twurlrc/TwurlRc.kt
package com.baulsupp.oksocial.services.twitter.twurlrc

import com.baulsupp.oksocial.services.twitter.TwitterCredentials

class TwurlRc {
    var profiles: Map<String, Map<String, TwitterCredentials>>? = null
    var configuration: Map<String, List<String>>? = null

    fun defaultProfile(): List<String> {
        return configuration!!["default_profile"]!!
    }

    fun readCredentials(username: String, consumerKey: String): TwitterCredentials {
        return profiles!![username]!!.get(consumerKey)!!
    }
}
